great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1122 @@
+        assert isinstance(self.ignored_columns, list)
+
+        self.excluded_expectations = excluded_expectations or []
+        assert isinstance(self.excluded_expectations, list)
+
+        self.value_set_threshold = value_set_threshold
+        assert isinstance(self.value_set_threshold, str)
+
+        self.not_null_only = not_null_only
+        assert isinstance(self.not_null_only, bool)
+
+        self.table_expectations_only = table_expectations_only
+        assert isinstance(self.table_expectations_only, bool)
+
+        self.primary_or_compound_key = primary_or_compound_key or []
+        assert isinstance(self.primary_or_compound_key, list)
+
+        if self.table_expectations_only:
+            self.ignored_columns = self.dataset.get_table_columns()
+
+        if self.primary_or_compound_key:
+            for column in self.primary_or_compound_key:
+                if column not in dataset.get_table_columns():
+                    raise ValueError(
+                        f"Column {column} not found. Please ensure that this column is in the dataset if"
+                        f"you would like to use it as a primary_or_compound_key."
+                    )
+
+        included_columns = [
+            column_name
+            for column_name in dataset.get_table_columns()
+            if column_name not in self.ignored_columns
+        ]
+
+        for column_name in included_columns:
+            self._add_column_cardinality_to_column_info(dataset, column_name)
+            self._add_column_type_to_column_info(dataset, column_name)
+
+        if self.semantic_type_dict is not None:
+            self._validate_semantic_types_dict(self.dataset)
+            for column_name in included_columns:
+                self._add_semantic_types_by_column_from_config_to_column_info(
+                    column_name
+                )
+        self.semantic_type_functions = {
+            "datetime": self._build_expectations_datetime,
+            "numeric": self._build_expectations_numeric,
+            "string": self._build_expectations_string,
+            "value_set": self._build_expectations_value_set,
+            "boolean": self._build_expectations_value_set,
+            "other": self._build_expectations_for_all_column_types,
+        }
+
+    def build_suite(self, tolerance=0):
+        """
+        User-facing expectation-suite building function. Works with an instantiated UserConfigurableProfiler object.
+        Args:
+            tolerance: The tolerance to use when adding expectations
+
+        Returns:
+            An expectation suite built either with or without a semantic_types dict
+
+        """
+        if len(self.dataset.get_expectation_suite().expectations) > 0:
+            suite_name = self.dataset._expectation_suite.expectation_suite_name
+            self.dataset._expectation_suite = ExpectationSuite(suite_name)
+
+        if self.table_expectations_only is True:
```

Comment:
I'm assuming there should be something here to make sure only table level expectations are set. I do love the logger message though, super helpful :)

Reply:
Great call-out! I moved this message to the __init__